Updated albert.md doc for ALBERT model

ENate opened the pull request:
# What does this PR do?
I updated the ```albert.md``` doc assigned to me by @stevhliu by adding the vital links and notes on how to use the ALBERT model.

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Fixes # (issue)


## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.

It was discussed on Github on the following link:
```
https://github.com/huggingface/transformers/issues/20055
```
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation). Yes
- [ ] Did you write any new necessary tests?

No
## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- text models: @ArthurZucker and @younesbelkada
- vision models: @amyeroberts
- speech models: @sanchit-gandhi
- graph models: @clefourrier

Library:

- flax: @sanchit-gandhi
- generate: @gante
- pipelines: @Narsil
- tensorflow: @gante and @Rocketknight1
- tokenizers: @ArthurZucker
- trainer: @muellerzr and @pacman100

Integrations:

- deepspeed: HF Trainer/Accelerate: @pacman100
- ray/raytune: @richardliaw, @amogkam
- Big Model Inference: @SunMarc
- quantization (bitsandbytes, autogpt): @SunMarc and @younesbelkada

Documentation: @stevhliu and @MKhalusova

HF projects:

- accelerate: [different repo](https://github.com/huggingface/accelerate)
- datasets: [different repo](https://github.com/huggingface/datasets)
- diffusers: [different repo](https://github.com/huggingface/diffusers)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Maintained examples (not research project or legacy):

- Flax: @sanchit-gandhi
- PyTorch: See Models above and tag the person corresponding to the modality of the example.
- TensorFlow: @Rocketknight1

 -->

ENate committed:
Updated albert.md doc for ALBERT model

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,81 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+- A blog post on how to train AlBERT with Blurr for sequence classification can be done by adapting this [example](https://huggingface.co/blog/fastai).
+- A blog post on how to use [Ray to tune hyperparameters](https://huggingface.co/blog/ray-tune) can be adapted for AlBERT.
+- A notebook on how to [Fine-tune ALBERT for sentence-pair classification](https://github.com/NadirEM/nlp-notebooks/blob/master/Fine_tune_ALBERT_sentence_pair_classification.ipynb)
+- A notebook on how to [Fine Tuning Transformer for MultiLabel Text Classification](https://github.com/abhimishra91/transformers-tutorials/blob/master/transformers_multi_label_classification.ipynb)
```
This can also be removed.

ENate replied:
So this example
```
- A notebook on how to [Fine-tune ALBERT for sentence-pair classification](https://github.com/NadirEM/nlp-notebooks/blob/master/Fine_tune_ALBERT_sentence_pair_classification.ipynb). 🌎
```
should also  be removed as well because it is not from the community?

stevhliu replied:
That one is okay since it's linked from the [Community resources](https://huggingface.co/docs/transformers/community) doc.

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,81 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+
+- A blog post on how to train AlBERT with Blurr for sequence classification can be done by adapting this [example](https://huggingface.co/blog/fastai).
+- A blog post on how to use [Ray to tune hyperparameters](https://huggingface.co/blog/ray-tune) can be adapted for AlBERT.
+- A notebook on how to [Fine-tune ALBERT for sentence-pair classification](https://github.com/NadirEM/nlp-notebooks/blob/master/Fine_tune_ALBERT_sentence_pair_classification.ipynb)
```
```suggestion
- A notebook on how to [Fine-tune ALBERT for sentence-pair classification](https://github.com/NadirEM/nlp-notebooks/blob/master/Fine_tune_ALBERT_sentence_pair_classification.ipynb). 🌎
```

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,81 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+- A notebook on how to [Fine-tune ALBERT for sentence-pair classification](https://github.com/NadirEM/nlp-notebooks/blob/master/Fine_tune_ALBERT_sentence_pair_classification.ipynb)
+- A notebook on how to [Fine Tuning Transformer for MultiLabel Text Classification](https://github.com/abhimishra91/transformers-tutorials/blob/master/transformers_multi_label_classification.ipynb)
+
+- Adapt this blog post on [Getting Started with Sentiment Analysis using Python](https://huggingface.co/blog/sentiment-analysis-python) for AlBERT.
```
Also remove

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,81 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+- Adapt this blog post on [Getting Started with Sentiment Analysis using Python](https://huggingface.co/blog/sentiment-analysis-python) for AlBERT.
+
+- [`AlBertForSequenceClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/text-classification).
+- [`TFAlBertForSequenceClassification`] can be done by adapting this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/text-classification).
```
```suggestion
- [`TFAlBertForSequenceClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/text-classification).
```

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,81 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+<PipelineTag pipeline="token-classification"/>
+
+- [`AlBertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/token-classification) and can be implemented by adapting the example in this [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification.ipynb).
+- [`TFAlBertForTokenClassification`] is supported by adapting this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/token-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification-tf.ipynb).
```
```suggestion
- [`TFAlBertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/token-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification-tf.ipynb).
```

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,81 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+- [`AlBertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/token-classification) and by adapting the [ example in this notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification.ipynb).
+- [`TFAlBertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/token-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification-tf.ipynb).
+- [`FlaxAlBertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/flax/token-classification).
+- [`FlaxAlBertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/flax/token-classification).
```
Duplicate ;)

```suggestion
```

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_27223). All of your documentation changes will be reflected on that endpoint.

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,81 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+
+- Check the  [Multiple choice task guide](../tasks/multiple_choice) on how to use the model.
+
+⚗️ Optimization
+
+- Blog posts on methods and tools [ for efficient training on a single GPU
+](https://huggingface.co/docs/transformers/perf_train_gpu_one), [efficient Inference on a Single GPU](https://huggingface.co/docs/transformers/perf_infer_gpu_one) and [efficient Inference on CPU
+](https://huggingface.co/docs/transformers/perf_infer_cpu) are also available.
+- A blog post on how [Optimizing Transformers for GPUs with 🤗 Optimum](https://www.philschmid.de/optimizing-transformers-with-optimum-gpu) can be modified for ALBERT.
+- A blog post on [Optimizing Transformers with Hugging Face Optimum](https://huggingface.co/blog/optimum-inference) and [an example can be found here](https://www.philschmid.de/optimizing-transformers-with-optimum).
```
Let's also remove these sections since they aren't directly applicable to ALBERT

ENate replied:
But I saw that, in the following links,  training efficiency on CPUs is discussed for generic BERT-based models:
```
⚗️ Optimization

- A blog post [ for efficient training on a single GPU
](https://huggingface.co/docs/transformers/perf_train_gpu_one).
```
- while ALBERT is also referenced here as well:

```
- A blog post on [Optimizing Transformers with Hugging Face Optimum](https://huggingface.co/blog/optimum-inference).
```

Do I still remove the entire 'Optimization' section?

stevhliu replied:
Yeah those are very generic and broad docs, so I think we should remove it entirely

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,81 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+- A blog post on [Optimizing Transformers with Hugging Face Optimum](https://huggingface.co/blog/optimum-inference) and [an example can be found here](https://www.philschmid.de/optimizing-transformers-with-optimum).
+
+
+⚡️ Inference
+
+- A blog post on how to [Accelerate BERT inference with Hugging Face Transformers and AWS Inferentia](https://huggingface.co/blog/bert-inferentia-sagemaker) can be adapted for AlBERT.
+- A blog post on [Serverless Inference with Hugging Face's Transformers, DistilBERT and Amazon SageMaker](https://www.philschmid.de/sagemaker-serverless-huggingface-distilbert) can be adapted for AlBERT.
```
Let's also remove these

ENate replied:
Done

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,81 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+🚀 Deploy
+
+- A blog post on how to [train ALBERT for natural language processing with TensorFlow on Amazon SageMaker](https://aws.amazon.com/blogs/machine-learning/train-albert-for-natural-language-processing-with-tensorflow-on-amazon-sagemaker/).
+- A post on how to [achieve 35% faster training with Hugging Face Deep Learning Containers on Amazon SageMaker](https://dataintegration.info/achieve-35-faster-training-with-hugging-face-deep-learning-containers-on-amazon-sagemaker).
```
Also remove since its not an official Hugging Face resource

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,81 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+
+- A blog post on how to [train ALBERT for natural language processing with TensorFlow on Amazon SageMaker](https://aws.amazon.com/blogs/machine-learning/train-albert-for-natural-language-processing-with-tensorflow-on-amazon-sagemaker/).
+- A post on how to [achieve 35% faster training with Hugging Face Deep Learning Containers on Amazon SageMaker](https://dataintegration.info/achieve-35-faster-training-with-hugging-face-deep-learning-containers-on-amazon-sagemaker).
+- Adapt the guide on this blog post on how to [Deploy BERT with Hugging Face Transformers, Amazon SageMaker and Terraform module](https://www.philschmid.de/terraform-huggingface-amazon-sagemaker) for ALBERT.
```
Also remove since its for DistilBERT

stevhliu reviewed (commented):
Thanks for working on this! 👏 

My main comment is lets keep these resources actionable and super easy-to-use (no adapting required) by only listing resources that directly work with ALBERT and if they're from an official Hugging Face source or by a Hugging Face team or community member.

ENate commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,81 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+
+<PipelineTag pipeline="text-classification"/>
+
+- A blog post on how to train AlBERT with Blurr for sequence classification can be done by adapting this [example](https://huggingface.co/blog/fastai).
```
Thanks. I can make the changes again. First time contribution for me. Is that okay?

stevhliu replied:
Yes of course, and welcome to the Hugging Face community! ❤️

ENate replied:
Thanks :)

ENate committed:
Update docs/source/en/model_doc/albert.md

Fixed Resources heading

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

ENate committed:
Update the ALBERT model doc resources

Fixed resource example for fine-tuning the ALBERT sentence-pair classification.

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

ENate committed:
Update docs/source/en/model_doc/albert.md

Removed resource duplicate

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -60,13 +60,80 @@ Next sentence prediction is replaced by a sentence ordering prediction: in the i
+
+- Adapt this blog post on [Getting Started with Sentiment Analysis using Python](https://huggingface.co/blog/sentiment-analysis-python) for AlBERT.
+
+- [`AlBertForSequenceClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/text-classification).
```
The "B" should be lowercase (this typo occurs several times so I would double check to make sure they're all lowercase)

```suggestion
- [`AlbertForSequenceClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/text-classification).
```

ENate replied:
Ok sure :)

ENate committed:
Updated albert.md doc with reviewed changes

ENate committed:
Updated albert.md doc for ALBERT

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -59,13 +59,69 @@ This model was contributed by [lysandre](https://huggingface.co/lysandre). This
+
+- [`AlbertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/token-classification).
+
+- [`TFAlbertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/token-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification-tf.ipynb).
```
Oops this is a duplicate of the `TFAlbertForTokenClassification` above!

```suggestion
```

ENate replied:
Thanks again :) I promise to do better next time :)

stevhliu reviewed (commented):
Nice job, just one more change and we're ready to merge!

ENate committed:
Update docs/source/en/model_doc/albert.md

Removed duplicates from  updated docs/source/en/model_doc/albert.md

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

stevhliu commented on docs/source/en/model_doc/albert.md:
```diff
@@ -59,13 +59,68 @@ This model was contributed by [lysandre](https://huggingface.co/lysandre). This
+- [`TFAlbertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/token-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification-tf.ipynb).
+
+
+- [`AlbertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/token-classification).
```
This is also a duplicate ;)

stevhliu replied:
Hi @ENate, would you mind removing the following duplicate line, then we can get your first PR merged!

```suggestion
```

stevhliu reviewed (approved):
Thanks again for iterating and improving this PR! 👍

stevhliu committed:
Update docs/source/en/model_doc/albert.md

ENate reviewed (commented):
More updates for ALBERT doc

stevhliu commented:
Thanks again for your contribution!

ENate commented:
No worries :)

ENate commented:
Hi @stevhliu. Do you think it may be great to modify the docs in such a way that we can have multiple tabs for examples in Jax, Tensorflow or Pytorch or this is not necessary? For example, each tab will consist of examples written in a particular framework?

stevhliu commented:
Hi @ENate, we currently already do that where appropriate with framework-specific code blocks such as the ones seen [here](https://huggingface.co/docs/transformers/quicktour#use-another-model-and-tokenizer-in-the-pipeline) :)

ENate commented:
Okay right :) I was thinking about tabs for each framework. Like on clicking on the tab, someone using the docs can easily see the examples related to either TensorFlow or PyTorch. But thanks
